File: app/Item.php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Item extends Model
{
    protected $guarded = [];

    /**
     * The attributes that should be hidden for arrays.
     *
     * @var array
     */
    protected $hidden = [
        'created_at',
        'updated_at',
        'destination_id',
        'category_id'
    ];

    public function parent()
    {
        return $this->belongsTo(Item::class);
    }

    public function children()
    {
        return $this->hasMany(Item::class);
    }

    public function destination()
    {
        return $this->belongsTo(Destination::class);
    }

    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    public function orders()
    {
        return $this->belongsToMany(Order::class)
            ->using(OrderItem::class)
            ->withPivot('quantity');
    }
}

File: app/models/Item.py
""" User Model """

from masoniteorm.models import Model
from masoniteorm.relationships import belongs_to_many, belongs_to
from masoniteorm.scopes import SoftDeletesMixin
from app.models.Category import Category
from app.models.Destination import Destination


class Item(Model, SoftDeletesMixin):
  """Item Model"""

  __timestamps__ = False

  @belongs_to_many(
    'item_id',
    'order_id',
    with_fields=['quantity']
  )
  def order(self):
    from app.models.Order import Order
    return Order

  @belongs_to(
    'category_id',
    'id'
  )
  def category(self):
    return Category

  @belongs_to(
    'destination_id',
    'id'
  )
  def destination(self):
    return Destination

  pass
